WalletWasabi.Gui/Controls/WalletExplorer/ClosedWalletViewModel.cs

```diff
@@ -25,7 +25,7 @@ public ClosedWalletViewModel(Wallet wallet) : base(wallet)
 
 					await global.WalletManager.StartWalletAsync(Wallet);
 				}
-				catch (TaskCanceledException ex) when (ex is TaskCanceledException || ex is OperationCanceledException)
+				catch (OperationCanceledException ex)
```

Comment:
Good catch and a bit scary to me because `TaskCancelException` is the norm except in those case where we use `cancel.ThrowIfCancellationRequested()` as we don in this code.

Reply:
There is only one derivation. This will catch only OperationCanceledException and TaskCanceledException.
Derived System.Threading.Tasks.TaskCanceledException

https://docs.microsoft.com/en-us/dotnet/api/system.operationcanceledexception?view=netcore-3.1